pkg/cmd/pulumi/new_test.go

```diff
@@ -1019,3 +1021,78 @@ func TestGenerateOnlyProjectCheck(t *testing.T) {
 		})
 	}
 }
+
+func TestPulumiNewConflictingProject(t *testing.T) {
+	t.Parallel()
+
+	b := &backend.MockBackend{
+		DoesProjectExistF: func(ctx context.Context, orgName string, projectName string) (bool, error) {
+			if projectName == "existing-project-name" {
+				return true, nil
+			}
+			return false, nil
+		},
+	}
+
+	assert.NoError(t,
+		validateProjectNameInternal(
+			context.Background(), b, "moolumi", "some-project-name", false /* generateOnly */, display.Options{},
+			func(s string) error {
+				assert.Fail(t, "this should not be called as this is a not a duplicate project name")
+				return nil
+			},
+		))
+
+	var called bool
+	assert.NoError(t,
+		validateProjectNameInternal(
+			context.Background(), b, "moolumi", "existing-project-name", false /* generateOnly */, display.Options{},
+			func(s string) error {
+				called = true
+				return nil
+			},
+		))
+	assert.Truef(t, called, "expected resolution to be called with duplicate name")
+}
+
+func TestPulumiNewConflictingStack(t *testing.T) {
+	t.Parallel()
+
+	b := &backend.MockBackend{}
+	s := &backend.MockStack{
+		RefF: func() backend.StackReference {
+			return &backend.MockStackReference{
+				StringV: "stack",
+				NameV:   "stack",
+			}
+		},
+		BackendF:              func() backend.Backend { return b },
+		DefaultSecretManagerF: func(info *workspace.ProjectStack) (secrets.Manager, error) { return nil, nil },
+	}
+
+	b.SupportsOrganizationsF = func() bool { return true }
+	b.CreateStackF = func(
+		ctx context.Context, ref backend.StackReference,
+		stackName string, opts *backend.CreateStackOptions,
+	) (backend.Stack, error) {
+		return nil, &backend.StackAlreadyExistsError{StackName: stackName}
+	}
+	b.GetStackF = func(ctx context.Context, stackRef backend.StackReference) (backend.Stack, error) {
+		return s, nil
+	}
+
+	prompt := func(yes bool, valueType string, defaultValue string, secret bool,
+		isValidFn func(value string) error, opts display.Options,
+	) (string, error) {
+		return "stack", nil
+	}
+
+	_, err := promptAndCreateStackInternal(context.Background(), b, prompt, "", "root", true, true, display.Options{
+		Color:         cmdutil.GetGlobalColorization(),
+		IsInteractive: true,
+	}, "", func(stackName string, opts display.Options) (bool, error) {
+		return true, nil
+	})
+
+	assert.NoError(t, err)
```

Comment:
Don't feel like this test is testing enough, should we not check that GetStack is called for the correct stack ref?

Reply:
Added

```go
assert.Equal(t, "stack", stackRef.FullyQualifiedName().String())
```